pkg/collector/corechecks/snmp/report/report_bandwidth_usage_test.go

```diff
@@ -0,0 +1,395 @@
+					// ifHCInOctets
+					"1.3.6.1.2.1.31.1.1.1.6": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 5000000.0,
+						},
+					},
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+					// ifHighSpeed
+					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 80.0,
+						},
+					},
+				},
+			},
+			[]Metric{
+				// ((5000000 * 8) / (80 * 1000000)) * 100 = 50.0
+				{"snmp.ifBandwidthInUsage.rate", 50.0},
+			},
+			nil,
+		},
+		{
+			"snmp.ifBandwidthOutUsage.Rate submitted",
+			checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.10", Name: "ifHCOutOctets"},
+			"9",
+			&valuestore.ResultValueStore{
+				ColumnValues: valuestore.ColumnResultValuesType{
+					// ifHCInOctets
+					"1.3.6.1.2.1.31.1.1.1.6": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 5000000.0,
+						},
+					},
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+					// ifHighSpeed
+					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 80.0,
+						},
+					},
+				},
+			},
+			[]Metric{
+				// ((1000000 * 8) / (80 * 1000000)) * 100 = 10.0
+				{"snmp.ifBandwidthOutUsage.rate", 10.0},
+			},
+			nil,
+		},
+		{
+			"not a bandwidth metric",
+			checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.99", Name: "notABandwidthMetric"},
+			"9",
+			&valuestore.ResultValueStore{
+				ColumnValues: valuestore.ColumnResultValuesType{},
+			},
+			[]Metric{},
+			nil,
+		},
+		{
+			"missing ifHighSpeed",
+			checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.6", Name: "ifHCInOctets"},
+			"9",
+			&valuestore.ResultValueStore{
+				ColumnValues: valuestore.ColumnResultValuesType{
+					// ifHCInOctets
+					"1.3.6.1.2.1.31.1.1.1.6": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 5000000.0,
+						},
+					},
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+				},
+			},
+			[]Metric{},
+			fmt.Errorf("bandwidth usage: missing `ifHighSpeed` metric, skipping metric. fullIndex=9"),
+		},
+		{
+			"missing ifHCInOctets",
+			checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.6", Name: "ifHCInOctets"},
+			"9",
+			&valuestore.ResultValueStore{
+				ColumnValues: valuestore.ColumnResultValuesType{
+					// ifHCOutOctets
+					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 1000000.0,
+						},
+					},
+					// ifHighSpeed
+					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
+						"9": {
+							Value: 80.0,
+						},
+					},
+				},
+			},
+			[]Metric{},
+			fmt.Errorf("bandwidth usage: missing `ifHCInOctets` metric, skipping this row. fullIndex=9"),
+		},
+		{
+			"missing ifHCOutOctets",
```

Comment:
I don't understand this test. Isn't `1.3.6.1.2.1.31.1.1.1.6` the `ifHCInOctets` oid?

Reply:
in this test:

```

		{
			"missing ifHCOutOctets",
			checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.6", Name: "ifHCOutOctets"},
			"9",
			&valuestore.ResultValueStore{
				ColumnValues: valuestore.ColumnResultValuesType{
					// ifHCOutOctets
					"1.3.6.1.2.1.31.1.1.1.10": map[string]valuestore.ResultValue{
						"9": {
							Value: 1000000.0,
						},
					},
					// ifHighSpeed
					"1.3.6.1.2.1.31.1.1.1.15": map[string]valuestore.ResultValue{
						"9": {
							Value: 80.0,
						},
					},
				},
			},
			[]Metric{},
			fmt.Errorf("bandwidth usage: missing `ifHCOutOctets` metric, skipping this row. fullIndex=9"),
		},
```

`checkconfig.SymbolConfig{OID: "1.3.6.1.2.1.31.1.1.1.6", Name: "ifHCOutOctets"},` is the symbol from profiles or metrics configs

But `valuestore.ResultValueStore` doesn't contain `1.3.6.1.2.1.31.1.1.1.6`, that's why the `"missing ifHCOutOctets"`

Reply:
I see, so you are changing the `name` in the oid symbol.

Reply:
Opinionated, it's misleading as `1.3.6.1.2.1.31.1.1.1.6` is `ifHCInOctets` OID. Again opinionated.